core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -356,6 +356,90 @@ public void write(byte[] b, int off, int len) throws IOException {
         writeBytes(b, off, len);
     }
 
+    /**
+     * Uses {@link StreamOutput#writeVInt(int)} to first provide the size, then it writes the array using each instances
+     * {@link Writeable#writeTo(StreamOutput)} method.
+     * <p>
+     * This is meant to replace boilerplate surrounding code of the form:
+     * <pre>
+     * YourObject[] objects = ...;
+     * out.writeVInt(objects.length);
+     * for (YourObject object : objects) {
+     *     object.writeTo(out);
+     * }
+     * </pre>
+     *
+     * @param array The array to write
+     */
+    public <T extends Writeable> void writeArray(T[] array) throws IOException {
+        writeVInt(array.length);
```

Comment:
Same comment on the other side - use writeList, probably. Either convert to a list when calling or just convert the object to storing a list rather than an array.